### PR TITLE
Improve empty requests treatment in geocoders

### PR DESCRIPTION
**Pull request by ethervoid**
* Fix for Mapbox geocoder to handle empty requests and empty responses
* Remove raising an exception when non parameters are passed to the HERE geocoder
* Fix for HERE geocoder with non empty requests
* Added more coverage to the google geocoder credentials parse logic

### CODE REVIEW

File: server/lib/python/cartodb_services/cartodb_services/mapbox/geocoder.py [hunk 1]
```diff
@@ -60,7 +60,10 @@ def _extract_lng_lat_from_feature(self, feature):
     @qps_retry(qps=10)
     def geocode(self, searchtext, city=None, state_province=None,
                 country=None):
-        address = [searchtext]
+        if not searchtext:
```
Comment (antoniocarlon) [inline server/lib/python/cartodb_services/cartodb_services/mapbox/geocoder.py]: we could add a check for `searchtext.strip()` here

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+January 29th, 2018
```
Comment (antoniocarlon) [inline NEWS.md]: We could modify the date here

File: server/lib/python/cartodb_services/cartodb_services/mapbox/geocoder.py (hunk 1 above)
Comment (ethervoid) [reply server/lib/python/cartodb_services/cartodb_services/mapbox/geocoder.py]: 👍

**Comment by ethervoid** [comment]
@antoniocarlon time for a second round :)

**Pull request merged**